src/nrnoc/multicore.c

```diff
@@ -677,7 +677,7 @@ void nrn_threads_free() {
 			spDestroy(nt->_sp13mat);
 			nt->_sp13mat = 0;
 		}
-		if(nt->_nrn_fast_imem) { fast_imem_free(); }
+		nt->_nrn_fast_imem = NULL;
```

Comment:
I see that it does not need to be freed here because it will be freed when the number of threads changes or fast imem is turned off. However, what are the circumstances in which it will not be freed since the call of the nrn_threads_free function inherently means that the number of threads is changing.  In other words, what are the circumstances in which someone accesses the fast_imem storage after a call to this function?

Reply:
I noticed that there was an issue with our `Simulation Stack` tests that included the `fast membrane current` calculation with the addition of `fast_imem_free()`. To be more specific, the values of the `membrane current` reports were all zero, instead of the correct non-zero value.
I haven't managed to find the reason why freeing the `fast_imem` struct is problematic since from what I've seen the `fast_imem` struct is reallocated before running the simulation.
I would suggest handling this as a separate issue, as it is not related to the `cmake` build system, and go ahead with the review of this branch. I will create a ticket with more details about the issue and we can follow it up there.
Finally, I tested our `SimulationStack` CI with the current state of this branch using the `cmake` build system and it is passing, so from me we can go ahead merging this branch.